app/components/App.js

```diff
@@ -34,22 +34,28 @@ export default class App extends Component {
   handleImageData(inputState){
     let content
     let statePromise = new Promise((resolve, reject)=>{
-      console.log('loading...')
       this.setState({
         error: false,
         apiResults: [],
         compareResults: [],
         loadingStatus: true,
         imagePreviewUrl: inputState.imagePreviewUrl,
       })
-      setTimeout(function(){
-        resolve('promise resolved');
-      }, 1100);
+
+      setTimeout(()=>{
+        if(this.state.imagePreviewUrl && this.state.loadingStatus){
+          console.log('loading...')
```

Comment:
I don't think I need to keep them in there, I'll remove them upon the next refactor.